cocos/core/math/color.ts

```diff
@@ -42,23 +42,27 @@ const A_INDEX = 3;
 const mathAbs = Math.abs;
 const mathMax = Math.max;
 
+function freezeColor (r: number, g: number, b: number, a: number): Readonly<Color> {
+    return Object.freeze(new Color(r, g, b, a));
+}
+
 /**
  * @en Representation of RGBA colors.<br/>
  * Each color component is an integer value with a range from 0 to 255.<br/>
  * @zh 通过 Red、Green、Blue 颜色通道表示颜色，并通过 Alpha 通道表示不透明度。<br/>
  * 每个通道都为取值范围 [0, 255] 的整数。<br/>
  */
 export class Color extends ValueType implements Modifiable {
-    public static WHITE = Object.freeze(new Color(255, 255, 255, 255));
-    public static GRAY = Object.freeze(new Color(127, 127, 127, 255));
-    public static BLACK = Object.freeze(new Color(0, 0, 0, 255));
-    public static TRANSPARENT = Object.freeze(new Color(0, 0, 0, 0));
-    public static RED = Object.freeze(new Color(255, 0, 0, 255));
-    public static GREEN = Object.freeze(new Color(0, 255, 0, 255));
-    public static BLUE = Object.freeze(new Color(0, 0, 255, 255));
-    public static CYAN = Object.freeze(new Color(0, 255, 255, 255));
-    public static MAGENTA = Object.freeze(new Color(255, 0, 255, 255));
-    public static YELLOW = Object.freeze(new Color(255, 255, 0, 255));
+    public static WHITE = freezeColor(255, 255, 255, 255);
+    public static GRAY = freezeColor(127, 127, 127, 255);
+    public static BLACK = freezeColor(0, 0, 0, 255);
+    public static TRANSPARENT = freezeColor(0, 0, 0, 0);
+    public static RED = freezeColor(255, 0, 0, 255);
+    public static GREEN = freezeColor(0, 255, 0, 255);
+    public static BLUE = freezeColor(0, 0, 255, 255);
+    public static CYAN = freezeColor(0, 255, 255, 255);
+    public static MAGENTA = freezeColor(255, 0, 255, 255);
+    public static YELLOW = freezeColor(255, 255, 0, 255);
```

Comment:
Can just assign Object.freeze to a variable? Such as
```
const freezeColor = Object.freeze;
```

Reply:
Probably we could, I didn't test that.
This depends on whether the implementation of `Object.freeze` uses `this` object since we change this object by `const freezeColor = Object.freeze`. 

For safety, it's better to use `const freezeColor = Object.freeze.bind(Object);`.

Reply:
After thinking a while, I think we should not use `const freezeColor = Object.freeze` since that will generate more code.

```ts
public static BLUE = freezeColor(new Color(0, 0, 255, 255)); 
```

will be mangled to 

Color.BLUE = aa(new Color(0, 0, 255, 255));

while
```ts
public static BLUE = freezeColor(0, 0, 255, 255);
```

will be mangled to 

Color.BLUE = aa(0, 0, 255, 255);

Reply:
ok